### PR TITLE
Code cleanup: Fixed (most) ES Lint issues

### PR DESCRIPTION
**Pull request by drayfarley**
### Summary 

Fixed issues noted by ES Lint. Most changes were adding props validation in .jsx files.
In addition, the outdated use of React.findDOMNode was replaced by the proper convention

### CODE REVIEW

File: server/methods.js [hunk 1]
```diff
@@ -523,10 +524,10 @@ function defaultDict() {
...
-  }
-}
\ No newline at end of file
+// function isJSON(str) {
+//   try {
+//     return (JSON.parse(str) && !!str);
+//   } catch (e) {
+//     return false;
+//   }
+// }
```
Comment (SamChou19815) [inline server/methods.js]: Nits: Remove? Is it still useful?

File: server/publications.js [hunk 2]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable meteor/audit-argument-checks */
```
Comment (SamChou19815) [inline server/publications.js]: 😮

File: imports/ui/Form.jsx [hunk 3]
```diff
@@ -169,10 +180,10 @@ export default class Form extends Component {
...
       if (error || result === 1) {
         // Success, so reset form
-        ReactDOM.findDOMNode(this.refs.ratingSlider).value = 3;
-        ReactDOM.findDOMNode(this.refs.diffSlider).value = 3;
-        ReactDOM.findDOMNode(this.refs.workloadSlider).value = 3;
-        ReactDOM.findDOMNode(this.refs.profSelect).value = "none";
+        this.ratingSlider.current.value = 3;
+        this.diffSlider.current.value = 3;
+        this.workloadSlider.current.value = 3;
+        this.profSelect.current.value = "none";
```
Comment (SamChou19815) [inline imports/ui/Form.jsx]: This is a good first step, but I still believe these types of imperative DOM manipulation can be eliminated. See https://reactjs.org/docs/refs-and-the-dom.html

Comment (julian-londono) [reply imports/ui/Form.jsx]: That's true, I believe we can take advantage of reactive variables like state variables but good start on this.

**Comment by julian-londono** [comment]
Creating a review app to check out before merging, will update link in a bit.

**Comment by julian-londono** [comment]
Deployed at: https://cornellreviews-dev-pr-99.herokuapp.com/

File: server/publications.js (hunk 2 above)
Comment (drayfarley) [reply server/publications.js]: Basically that checks for if the arguments for Meteor methods are "checked" but the docs said you should only care if you're using the Meteor checks package, which we aren't.

https://github.com/dferber90/eslint-plugin-meteor/blob/master/docs/rules/audit-argument-checks.md

Comment (SamChou19815) [reply server/publications.js]: @draycfarley Then it might be more helpful to disable it in `.eslintrc.json`>

Comment (julian-londono) [reply server/publications.js]: Moved it to eslintrc in latest commit @SamChou19815

**Pull request merged**